Azure.ResourceManager.Core/Resources/ResourceIdentifier.cs

```diff
@@ -60,7 +57,8 @@ public override int GetHashCode()
 
         public static implicit operator ResourceIdentifier(string other)
         {
-            return new ResourceIdentifier(other);
+            var val = other == null ? throw new ArgumentNullException(nameof(other)) : new ResourceIdentifier(other);
```

Comment:
for style reasons can we change this to an if throw otherwise assign?
```suggestion
            if(other == null)
                throw new ArgumentNullException(nameof(other));
            var val = new ResourceIdentifier(other);
```

Reply:
I will add null checks with the outcome of #118